### PR TITLE
Fixes for #28 SaveSimulationToXML destroys unicode characters

### LINKED ISSUE #28: SaveSimulationToXML destroys unicode characters.
See https://github.com/Open-Systems-Pharmacology/R-Toolbox/issues/54

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #28

### CODE REVIEW

File: src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp [hunk 1]
```diff
@@ -243,8 +243,11 @@ void SimModelComp_XMLHelper::SaveStringToFile(const std::string & str, const str
 		//implemented workaround via wstring as found here: 
 		//http://mariusbancila.ro/blog/2008/10/20/writing-utf-8-files-in-c/
 
-		wstring wstr(str.begin(), str.end());
-		string utf8string = to_utf8(wstr);
+        int size_needed = (int)str.size() + 1;
```
[the remark this answers is not in the document]
Comment (msevestre) [reply src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp]: Also remove the comment with the link above if it does not apply anymore

[the remark this answers is not in the document]
Comment (msevestre) [reply src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp]: Also would it be possible to add a unit test for that?

[the remark this answers is not in the document]
Comment (PavelBal) [reply src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp]: > Also indentation seems to be off

Indentation of the whole file does not correspond to the conding standards...

> Also would it be possible to add a unit test for that?

I am still not so familiar with the unit tests concept in c++, so no, I cannot do this right now. The unicode problem is kind of covered by the unit tests for the R-toolbox.

**Comment by msevestre** [comment]
@Yuri05 Can you please review and merge this changes?

File: src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp [hunk 2]
```diff
@@ -226,31 +226,36 @@ std::string to_utf8(const std::wstring& str)
...
+   {
+      //just writing std::string to file destroys UTF8 encoding
+      //implemented workaround via wstring as found here: 
+      //http://mariusbancila.ro/blog/2008/10/20/writing-utf-8-files-in-c/
+      //Converstion of input string to wchar as described here:
+      //https://stackoverflow.com/questions/29847036/convert-char-to-wchar-t-using-mbstowcs-s
+
+      int size_needed = (int)str.size() + 1;
+      size_t outSize;
+      wchar_t* outW = new wchar_t[size_needed];
```
Comment (Yuri05) [inline src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp]: looks like a memory leak; `outW `is not deleted

Comment (PavelBal) [reply src/OSPSuite.SimModelComp/src/SimModelComp_XMLHelper.cpp]: Garbage collector, I miss you...

**Pull request merged**